src/hgwosca.c

```diff
@@ -20,18 +20,17 @@
    Initialise population of `wolf_count` wolves, each with `dim` dimensions, where
    each dimension is bound by `min_positions` and `max_positions`.
  */
-double *gwo_init_population(size_t wolf_count,
+double *gwo_init_population(double* const population,
```

Comment:
shouldn't the return type be void, since you're not returning anything anymore? 

same with few others too

Reply:
Oh yeah thanks.

Reply:
wait, does C allow to have a function return type double and not return anything?

Reply:
Yeah it would seem so.

Reply:
the return type was double*, maybe it was a taking the no return as void* and that is quite flexible. but I'm not really sure.